.travis.yml upgrade

Marqin opened the pull request:
Now it test 6 builds:
- Ubuntu 14.04 -  x11 (g++)
- Ubuntu 14.04  - x11 (clang++)
- Ubuntu 14.04 - windows (via MinGW)
- OS X - osx (g++)
- OS X - osx (clang++)
- OS X - android
- OS X - iphone

Instead of just g++ on Ubuntu 12.04...

Commit:
upgrade .travis.yml

Now it test 6 builds:

* Ubuntu 14.04 -  x11 (g++)
* Ubuntu 14.04  - x11 (clang++)
* Ubuntu 14.04 - windows (via MinGW)
* OS X - osx (g++)
* OS X - osx (clang++)
* OS X - android
* OS X - iphone

Instead of just g++ on Ubuntu 12.04...

Marqin commented:
Build of last commit on this branch - https://travis-ci.org/Marqin/godot/builds/98370163

reduz commented:
awesome, how about Android and iOS? :)

On Tue, Dec 22, 2015 at 3:39 PM, Hubert Jarosz <notifications@github.com>
wrote:

> Build of last commit on this branch -
> https://travis-ci.org/Marqin/godot/builds/98370163
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/godotengine/godot/pull/3120#issuecomment-166699101>.
>

punto- commented:
what is this? how does it happen? do the builds actually run somewhere?

On 22 December 2015 at 15:46, Juan Linietsky <notifications@github.com>
wrote:

> awesome, how about Android and iOS? :)
>
> On Tue, Dec 22, 2015 at 3:39 PM, Hubert Jarosz <notifications@github.com>
> wrote:
>
> > Build of last commit on this branch -
> > https://travis-ci.org/Marqin/godot/builds/98370163
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/godotengine/godot/pull/3120#issuecomment-166699101>.
>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/godotengine/godot/pull/3120#issuecomment-166701028>.
>

Marqin commented:
@reduz I added iOS build on OS X - https://travis-ci.org/Marqin/godot/builds/98395485
@punto- It's Travis CI building my fork of Godot. Builds run on Travis CI servers. They are free for Open Source projects - https://travis-ci.com/plans

Marqin commented:
Adding android should be also simple, but I don't know yet how to install needed SDKs and NDKs via command line ( I'd done once that via Eclipse :laughing: ).

Marqin commented:
Adding android should be also simple, but I don't know yet how to install needed SDKs and NDKs via command line ( I'd done once that via Eclipse :laughing: ).

It could be easier if Travis CI supported multi-language projects or supported building Android projects ( which have preinstalled SDK/NDK ) on OS X ( then this would be tagged as android, not c++ ).

reduz commented:
good question, i never tried to add android compilation with apt, but i'm
not sure if it's possible because it auto updates itself

On Tue, Dec 22, 2015 at 5:50 PM, Hubert Jarosz <notifications@github.com>
wrote:

> Adding android should be also simple, but I don't know yet how to install
> needed SDKs and NDKs via command line ( I'd done once that via Eclipse [image:
> :laughing:] ).
>
> It could be easier if Travis CI supported multi-language projects or
> supported building Android projects ( which have preinstalled SDK/NDK ) on
> OS X ( then this would be tagged as android, not c++ ).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/godotengine/godot/pull/3120#issuecomment-166726458>.
>

Marqin commented:
It's not in APT, but I'll try with building Android on OS X - NDK and SDK in Homebrew repo.

Hmm, it looks like iPhone build failed on linking stage. Could you check it? https://travis-ci.org/Marqin/godot/jobs/98395490
Internet is only full of #define/-D workarounds for that.

Marqin commented:
@reduz Android builds OK on OS X Travis CI :+1:  So only problem is now with this iOS error.

vnen commented:
No Windows? Not even with MinGW?

(Also, I hope this commit history is cleaned up before merging).

Marqin commented:
@vnen 
No Windows (yet) support in Travis CI. Juan asked for help with CI on FB group, so I'd enhanced .travis.yml that's in Godot tree. In case someone wants to use it for their fork or you want to use it upstream.

I'll squash commits in the end.

Marqin commented:
If you want to have Windows CI and don't want to host it on your servers then your only choice is [AppVeyor](http://www.appveyor.com/).

Marqin commented:
@vnen I'd looked up and the only hosted CI with Windows support does not support Linux/OS X - [AppVeyor](http://www.appveyor.com/).

So if you want Windows, you need to self-host some buildbot/Jenkins or use two hosted CIs for different platforms.

reduz commented:
getting mingw with apt makes it very easy to build on windowd with travis ci


On Tue, Dec 22, 2015 at 6:59 PM, Hubert Jarosz <notifications@github.com>
wrote:

> @vnen <https://github.com/vnen> I'd looked up and the only hosted CI with
> Windows support does not support Linux/OS X - AppVeyor
> <http://www.appveyor.com/>.
>
> So if you want Windows, you need to self-host some buildbot/Jenkins or use
> two hosted CIs for different platforms.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/godotengine/godot/pull/3120#issuecomment-166741681>.
>

Marqin commented:
@reduz thanks. [It has successfully crosscompiled on Linux with MinGW.](https://travis-ci.org/Marqin/godot/jobs/98414199) Now the only problem is iPhone error.

Marqin commented:
@vnen I've squashed those commits into one. Also added MinGW build on Ubuntu.

@reduz This is ready for merge. iPhone failing build is godot bug and is fixed by #3124 ( [proof](https://travis-ci.org/Marqin/godot/jobs/98500762) ).

eska014 commented:
This was unintentionally closed from #3124 due to the string "fix #3120" in the opening comment.

akien-mga commented:
Merging, as apparently everything has been configured properly (and it's just one file anyway so it doesn't hurt if it needs further work in another PR).

Now the question is: how can we enable travis checks on the godotengine/godot repo? And maybe also on PRs?

akien-mga commented:
Should I register a "Godot" account at travis-ci.org and configure it from there?

reduz commented:
sure! I am not very familiar with it yet

On Thu, Dec 24, 2015 at 6:44 AM, Rémi Verschelde <notifications@github.com>
wrote:

> Should I register a "Godot" account at travis-ci.org and configure it
> from there?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/godotengine/godot/pull/3120#issuecomment-167078666>.
>

Marqin commented:
You have to register with your github account on travis-ci.org and turn this repo ON on repo page. Then from what I remember it will start building from next commit/new PR proposition

Marqin commented:
Tip: if you add [skip ci] to commit message it won't trigger build

neikeq commented:
- Go to https://travis-ci.org/ and _Sign in with Github_.
- Once you are in, go to your name (at the top right corner) and click _Accounts_. I guess the url would be https://travis-ci.org/profile/godotengine for the organization.
- You should see a list of the organization repos. If something is missing, click the _Sync_ button. Everything else is explained in the page:
![Quick Guide](https://cloud.githubusercontent.com/assets/7718100/11995097/07ab94ae-aa4b-11e5-891a-c1d3e89558a8.png)
After you toggled a repository, go to the repo's _Settings > Webhooks & Services_ on github. You should see Travis CI listed in the services panel.

vnen commented:
It seems only admins can do this. So @reduz or @punto- should set up Travis as @neikeq described.

akien-mga commented:
Indeed, I had a look and even though I'm part of the organization I can't toggle it on myself.

akien-mga commented:
@reduz Could you have a quick look at enabling travis-ci.org for @godotengine? See howto three comments above.

akien-mga commented:
travis-ci was enabled for the main repo, we should see it live next time a commit is pushed \o/

bojidar-bg commented:
@akien-mga ~ What about PR's?

akien-mga commented:
> What about PR's?

If I understand the doc correctly, it will only be triggered with new commits, not for the backlog. https://docs.travis-ci.com/user/getting-started/
